Source/Core/Core/HW/SI.cpp

```diff
@@ -234,9 +234,9 @@ void DoState(PointerWrap &p)
 		{
 			// If no movie is active, we'll assume the user wants to keep their current devices
 			// instead of the ones they had when the savestate was created.
-			if (!Movie::IsMovieActive())
-				return;
-
+			// However, change the devices anyways in case they add or remove
+			// a device after the savestate was created.
+			ChangeDeviceDeterministic(type, i);
```

Comment:
Oh, and you want to change to the current device->GetDeviceType(), not the type which was saved into the savestate.